### ISSUE TITLE
Failed to build on FreeBSD due to unconditional usage of splice() in sources

### ISSUE THREAD

**Issue by vgrebenschikov** [darkk/redsocks#110]
`
cc -g -O2 -I/usr/local/include -std=c99 -D_XOPEN_SOURCE=600 -D_DEFAULT_SOURCE -D_GNU_SOURCE -Wall   -c -o redsocks.o redsocks.c
redsocks.c:441:25: warning: implicit declaration of function 'splice' is invalid in C99 [-Wimplicit-function-declaration]
                        const ssize_t sent = splice(c->pisrc->read, NULL, out, NULL, avail, SPLICE_F_MOVE|SPLICE_F_NONBLOCK);
                                             ^
redsocks.c:441:72: error: use of undeclared identifier 'SPLICE_F_MOVE'
                        const ssize_t sent = splice(c->pisrc->read, NULL, out, NULL, avail, SPLICE_F_MOVE|SPLICE_F_NONBLOCK);
                                                                                            ^
`

**Comment by darkk** [comment]
Yep. Sorry, I have no permanent root access to machines running FreeBSD to test and develop for these platforms.
Was redsocks ever working for you? I've never tested it on FreeBSD and I'm unsure how to configure firewall for redsocks to receive TCP streams and UDP packets.

**Comment by PaulGWebster** [comment]
I could offer you a freebsd vm for playing about in to help get it working if required
